basho/riak_cs
# Handle object and file deletion

- New `riak_moss_delete_fsm` module coordinates deletion.
- `riak_moss_deleter` module performs actual deletion from `riak`. 
- `riak_moss_delete_fsm_eqc` module tests state transitions of the delete fsm.
- `riak_moss_deleter_test` tests for expected responses from the deleter module for a given input message.
### To Test
1. Create a moss bucket:
   `s3cmd mb s3://testbucket`
2. Put a file into the bucket:
   `s3cmd put <FILE_NAME_HERE> s3://testbucket/`
3. Verify the contents of the Riak buckets using erlang console for **Riak**:
   <pre>
   <code>
   {ok, C}=riak:local_client().
   C:list_keys(<<"objects:testbucket">>).
   C:list_keys(<<"blocks:testbucket">>).
   </code>
   </pre>
4. Delete the file on the **MOSS** console:
   `riak_moss_delete_fsm_sup:start_delete_fsm(node(), [<<"testbucket">>, <<"FILE_NAME_HERE">>, 300000]).`
5. Re-run the 2 `list_keys` calls from step 3 on the **Riak** console and each should now return empty lists (assuming there were no pre-existing files in the bucket).
